Content.Shared/_RMC14/Xenonids/Parasite/SharedXenoParasiteSystem.cs

```diff
@@ -321,21 +321,21 @@ public bool Infect(Entity<XenoParasiteComponent> parasite, EntityUid victim, boo
         if (!CanInfectPopup(parasite, victim, parasite, popup, force))
             return false;
 
-        if (_inventory.TryGetContainerSlotEnumerator(victim, out var slots, SlotFlags.MASK))
+        if (_inventory.TryGetContainerSlotEnumerator(victim, out var maskSlots, SlotFlags.MASK))
         {
-            var any = false;
-            while (slots.MoveNext(out var slot))
+            while (maskSlots.MoveNext(out var mask))
             {
-                if (slot.ContainedEntity != null)
-                {
-                    _inventory.TryUnequip(victim, victim, slot.ID, force: true);
-                    any = true;
-                }
+                if (mask.ContainedEntity != null)
+                    TryRipOffClothing(victim, mask.ContainedEntity.Value, mask.ID);
             }
+        }
 
-            if (any && _net.IsServer)
+        if (_inventory.TryGetContainerSlotEnumerator(victim, out var headSlots, SlotFlags.HEAD))
+        {
+            while (headSlots.MoveNext(out var head))
             {
-                _popup.PopupEntity(Loc.GetString("rmc-xeno-infect-success", ("target", victim)), victim);
+                if (head.ContainedEntity != null)
+                    TryRipOffClothing(victim, head.ContainedEntity.Value, head.ID);
```

Comment:
Won't this play two popups if you have both a mask and helmet? I'm not sure if it displays both in 13, but since we are doing popups here (like we should be) instead of chat messages, it would make these stack up. Might be easiest to have a doPopup boolean in TryRipOffClothing, and not play the helmet popup if the mask one happened.

Reply:
it does display both in 13 since it just prints to chat
Ok